### PR TITLE
feat(ui): Add new Collapsible component

### PR DESCRIPTION
**Pull request by matejminar**
Kapture 2020-12-14 at 17 12 55

We are using this pattern already in multiple places and each has its own implementation.
In the follow-up PR, I will replace all of them with this reusable one.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/collapsible.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    this.setState(prevState => ({
+      collapsed: !prevState.collapsed,
+    }));
+  };
+
+  renderCollapseButton() {
+    const {collapseButton} = this.props;
+
+    if (typeof collapseButton === 'function') {
+      return collapseButton({handleCollapse: this.onCollapseToggle});
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/collapsible.tsx:47]: We should swap the prefixes here, the callback should be `onCollapse` and the method should be `this.handleCollapseToggle`.

File: tests/js/spec/components/collapsible.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+describe('Collapsible', function () {
+  it('collapses items', function () {
+    const wrapper = mountWithTheme(<Collapsible>{items}</Collapsible>);
+
+    expect(wrapper.find('div').length).toBe(5);
+    expect(wrapper.find('div').at(2).text()).toBe('Item 3');
+
+    expect(wrapper.find('Button[data-test-id="expand"]').text()).toBe(
+      'Show 2 collapsed items'
+    );
```
Comment (billyvg) [inline tests/js/spec/components/collapsible.spec.jsx:19]: imo we shouldn't use `data-test-id` for Buttons, we should instead use `aria-label` as that'll build good habits to ensure that our buttons are accessible.

File: src/sentry/static/sentry/app/components/collapsible.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/components/collapsible.tsx:47]: good catch, will do

File: tests/js/spec/components/collapsible.spec.jsx (hunk 2 above)
Comment (matejminar) [reply tests/js/spec/components/collapsible.spec.jsx:19]: sure thing, let me switch to that

File: src/sentry/static/sentry/app/components/collapsible.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    const items = React.Children.toArray(children);
+    const canExpand = items.length > maxVisibleItems;
+    const itemsToRender =
+      collapsed && canExpand ? items.slice(0, maxVisibleItems) : items;
+    const numberOfCollapsedItems = items.length - itemsToRender.length;
+
+    return (
+      <React.Fragment>
+        {itemsToRender.map(item => item)}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/collapsible.tsx:86]: Do you need this map? There are no changes/extraction being done.

**Review by markstory — APPROVED** [review]

**Pull request merged**